bcdi/simulation/supportMaker.py

```diff
@@ -0,0 +1,414 @@
+
+    def set_planes(self, planes_list, plane_distance_origin_list):
+        """Set the planes."""
+        if len(planes_list) != len(plane_distance_origin_list):
+            print("the number of planes does not match the number of distances.")
+            sys.exit()
+
+        self.planes = np.array(planes_list)
+        self.planesDist = np.array(plane_distance_origin_list)
+
+        # based on voxel size - scale distance to plane metres to pixels
+        # convert existing plane distance
+        d_pix = np.sqrt(np.sum((self.planes) ** 2, axis=1))
+        print("\nD_pix", d_pix)
+
+        # convert existing plane distance to metres to user defined size
+        # with a scalefactor
+        d_m = np.sqrt(np.sum((self.vox_size * self.planes) ** 2, axis=1))
+        print("\nDM", d_m)
+
+        sf = (self.planesDist.reshape(1, self.planes.shape[0]) / d_m).reshape(
+            self.planes.shape[0], 1
+        )
+        self.scaled_planes = self.planes * sf
+
+        print(self.planesDist.reshape(1, self.planes.shape[0]))
+        print("\n###", self.scaled_planes)
+        print("\n### scale factor:", sf)
+        print("\n###", self.planes)
+        print("\n###", self.rawdata.shape)
+
+    def get_support(
+        self,
+    ):
+        """Return the generated support."""
+        return self.support
+
+    def get_planes(
+        self,
+    ):
+        """Return planes array."""
+        return self.planes
+
+    def get_planesDist(
+        self,
+    ):
+        """Return planes distance from origin."""
+        return self.planesDist
+
+    def calc_voxel_size(
+        self,
+        wavelength=None,
+        detDist=None,
+        pixel_size=None,
+        ang_step=None,
+        braggAng=None,
+    ):
+        """Calculate the voxel size."""
+        # use the experiment parameters to determine the voxel size
+        ss = self.support.shape
+
+        # calculate the angular dimension first
+        q = 4 * np.pi * np.sin(np.deg2rad(braggAng)) / wavelength
+        deltaQ = q * np.deg2rad(ang_step) * ss[0]
+        a = np.pi * 2 / deltaQ
+
+        # add the detector dimensions - don't forget to reverse -
+        # but pixels tend to be symmetric
+        pixel_size.reverse()
+        self.vox_size = np.r_[
+            a, wavelength * detDist / (np.array(ss[1:]) * np.array(pixel_size))
+        ]
+
+        print("Voxel dimensions: ", self.vox_size * 1e9, " (nm)")
+
+
+def generatePlanesCuboid(x, y, z):
+    """Make a cuboid of size x,y,z."""
+    planes = np.array(
+        [
+            [1, 0, 0],
+            [-1, 0, 0],
+            [0, 1, 0],
+            [0, -1, 0],
+            [0, 0, 1],
+            [0, 0, -1],
+        ]
+    )
+    planesDist = np.array(
+        [
+            [x / 2],
+            [x / 2],
+            [y / 2],
+            [y / 2],
+            [z / 2],
+            [z / 2],
+        ]
+    )
+    return planes, planesDist
+
+
+if __name__ == "__main__":
```

Comment:
I'm not sure about this part, don't you want to put it in a separate script? Ideally there are only functions and classes in a module, not executable code.